src/index.ts

```diff
@@ -146,15 +143,15 @@ export enum CommitTransferError {
   preimage_invalid,
 }
 
-export type CommitTransferResult = {
+export type CommitTransfersResult = {
   index: number,
-  error: CommitTransferError,
+  code: CommitTransferError,
 }
 
 export type AccountLookup = bigint // u128
 
-export type Command = CreateAccount | CreateTransfer | CommitTransfer | AccountLookup
-export type Result = CreateAccountResult | CreateTransferResult | CommitTransferResult | Account
+export type Command = Account | Transfer | Commit | AccountLookup
```

Comment:
We could do `Event` instead of `Command` to match the name of the type in the state machine, and because we use `command` for a different purpose in the replication protocol.

We could also do `AccountID` instead of `AccountLookup`.